src/pyproject_fmt/formatter/project.py

```diff
@@ -30,18 +30,18 @@ def fmt_project(parsed: TOMLDocument, conf: Config) -> None:
         opt_deps = cast(Table, project["optional-dependencies"])
         for value in opt_deps.values():
             normalize_pep508_array(cast(Array, value), conf.indent)
-        order_keys(opt_deps.value.body, (), sort_key=lambda k: k[0])
+        order_keys(opt_deps, (), sort_key=lambda k: k[0])
 
     for of_type in ("scripts", "gui-scripts", "entry-points", "urls"):
         if of_type in project:
             table = cast(Table, project[of_type])
```

Comment:
On a completely side-note here, you can also use `AbstractTable` in the places you are not sure if you are handling Table or InlineTable.